File: app/src/main/java/com/travlog/android/apps/viewmodels/outputs/MainViewModelOutputs.java
package com.travlog.android.apps.viewmodels.outputs;

import android.support.annotation.NonNull;

import com.travlog.android.apps.models.Note;

import java.util.List;

import io.reactivex.Observable;

public interface MainViewModelOutputs {

    @NonNull
    Observable<List<Note>> updateData();

    @NonNull
    Observable<Note> showNoteDetailsActivity();

    @NonNull
    Observable<Note> updateNote();

    @NonNull
    Observable<Integer> deleteNote();
}

File: app/src/main/java/com/travlog/android/apps/viewmodels/outputs/MainViewModelOutputs.kt
package com.travlog.android.apps.viewmodels.outputs

import com.travlog.android.apps.libs.rx.Optional
import com.travlog.android.apps.models.Note

import io.reactivex.Observable

interface MainViewModelOutputs {

    fun updateData(): Observable<List<Note>>

    fun showNoteDetailsActivity(): Observable<Note>

    fun clearNotes(): Observable<Optional<*>>

    fun updateNotes(): Observable<Note>

    fun deleteNote(): Observable<String>
}
